### ISSUE TITLE
Seems to ignore the param passed to flat.

### ISSUE THREAD

**Issue by abelta** [jonathantneal/array-flat-polyfill#1]
According to specification, flat(2) should perform a recursive flattening of the array up to two levels but This last part seems to not be working and be the same as flat().

**Comment by aaronbeall** [comment]
Yeah, this polyfill is just broken. `[[1], [2], [3]].flat()` gives `[[1], [2], 3]`...

**Comment by aaronbeall** [comment]
Fixed above issue in https://github.com/jonathantneal/array-flat-polyfill/pull/3

**Comment by jonathantneal** [comment]
Thanks, @aaronbeall! Hopefully we can get that in soon and publish an update. 😄

**Comment by jonathantneal** [comment]
I should have this published soon. https://codepen.io/jonneal/post/why-i-have-not-published

**Comment by jonathantneal** [comment]
I’ve run a performance metric between the original (bug fixed) `while` implementation and the `reduce` implementation. The `reduce` implementation is much faster.

I have also compared these with an additional optimization — one where we do not run the reducer if the `depth` is `0`. For this optimization, I can use either `Array.prototype.slice.call` or `Array.from`.

```js
	return depth ? Array.prototype.reduce.call(this, function (acc, cur) {
		if (Array.isArray(cur)) {
			acc.push.apply(acc, flat.call(cur, depth - 1));
		} else {
			acc.push(cur);
		}

		return acc;
	}, []) : Array.prototype.slice.call(this);
```

```js
	return depth ? Array.prototype.reduce.call(this, function (acc, cur) {
		if (Array.isArray(cur)) {
			acc.push.apply(acc, flat.call(cur, depth - 1));
		} else {
			acc.push(cur);
		}

		return acc;
	}, []) : Array.from(this);
```

In all of these cases, all browsers agree that the `while` loop us the slowest. However, they differ on which is the fastest.

https://www.browserscope.org/user/tests/table/ahBzfnVhLXByb2ZpbGVyLWhychELEgRUZXN0GICA0JTQ-M8LDA

| Browser | While | Reduce | Depth Reduce or `from` | Depth Reduce or `slice` |
| --- | --- | --- | --- | --- |
| Chrome | 4 (51% slower) | 2 (07% slower) | 3 (11% slower) | 1 |
| Firefox | 4 (34% slower) | 3 (19% slower) | 1 | 2 (10% slower) |
| Safari | 4 (29% slower) | 3 (09% slower) | 1 | 2 (03% slower) |

If I go with the fastest solution for Chrome, then Safari and Firefox would get a 3% - 10% slower version.
If I go with the fastest solution for Safari and Firefox, then Chrome would get an 11% slower version.

Since 11% > 10%, I will use the fastest solution for Chrome. This also happens to preserve IE9+ compatibility.

**Comment by jonathantneal** [comment]
This issue is resolved in 1.0.1, which has just been published.
